utils/src/main/kotlin/Logger.kt

```diff
@@ -34,3 +35,16 @@ val loggerOfClass = ConcurrentHashMap<Any, KotlinLogger>()
  */
 val <reified T : Any> T.log: KotlinLogger
     inline get() = loggerOfClass.getOrPut(T::class.java) { loggerOf(T::class.java) }
+
+private val KotlinLogger.statements by lazy { mutableSetOf<Triple<Any, Level, String>>() }
```

Comment:
This requires comparing every log statement to all others in this list, if you had used the type `MutableMap<Any, MutableMap<Level, String>>` the check would be more efficient. However, I'm fine with keeping in that way if the performance is still good. Could you do me a favor and rebase this branch on top of @fviernau's performance fix from today? Then I could make a test run for a large project to be on the safe side.

Reply:
> This requires comparing every log statement to all others in this set

I don't think this is the case: `mutableSetOf()` creates a `LinkedHashSet` which is baked by a `HashMap`. That is, `in` / `contains` should be very fast as it only looks up the hash for the `Triple`. Or am I missing something?

Reply:
Anyway, I've pushed an update with a `fixup` commit that shows how a nested-map-based solution could look like, for you to choose. Personally, provided that performance is indeed equal, I'd prefer the original solution as I find it to be more readable.

Reply:
That's true, I'd say drop the fixup commit then.

Reply:
Done.